src/composables/fungibleTokens.ts

```diff
@@ -74,6 +88,31 @@ export function useFungibleTokens() {
     return tokenBalances.value.filter((token) => token.address === address) || [];
   }
 
+  /**
+   * Extract unique tokens from the `tokenBalances` list and store them
+   * in the `customTokensAvailable` list.
+   */
+  function setCustomTokens() {
+    const uniqueTokens: IToken[] = uniqBy(tokenBalances.value, 'contractId')
+      .map((tokenBalance) => ({
+        contractId: tokenBalance.contractId,
+        protocol: tokenBalance.protocol,
+        name: tokenBalance?.name!,
+        symbol: tokenBalance?.symbol!,
+      }));
+
+    customTokensAvailable.value = uniqueTokens.reduce((customTokens, token) => {
```

Comment:
Maybe instead of reduce you could use `objectFrom` ?

Reply:
I don't think it makes sense to use `Object.fromEntries` here as `uniqueTokens ` and `customTokensAvailable` have different structures.
Did you mean something else with `objectFrom`?